query.go

```diff
@@ -381,13 +381,16 @@ func queryChainServiceBatch(
 			mtxPeerStates.Lock()
 			peerStates[sp.Addr()] = queryMsgs[handleQuery]
 			mtxPeerStates.Unlock()
+
+			exiting := false
```

Comment:
Non-blocking comment: as discussed offline, could make this be the query state instead so that we only update it in one place.